### ISSUE TITLE
SC2059 does not apply to "%s $1 \n"

### ISSUE THREAD

**Issue by raphaelahrens** [koalaman/shellcheck#2084]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2059
- My shellcheck version (`shellcheck --version` or "online"):  online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

printf "test $1\n"
printf "%s $1 \n" "test"
printf "%s ""$1 \n" "test"

```

#### Here's what shellcheck currently says:

```
Line 3:
printf "test $1\n"
       ^-- SC2059: Don't use variables in the printf format string. Use printf '..%s..' "$foo".
```

#### Here's what I wanted or expected to see:

```
Line 3:
printf "test $1\n"
       ^-- SC2059: Don't use variables in the printf format string. Use printf '..%s..' "$foo".
Line 4:
printf "%s $1 \n" "test"
       ^-- SC2059: Don't use variables in the printf format string. Use printf '..%s..' "$foo".
Line 5:
printf "%s ""$1 \n" "test"
       ^-- SC2059: Don't use variables in the printf format string. Use printf '..%s..' "$foo".
```

**Comment by koalaman** [comment]
This is currently intentional: ShellCheck does not emit this warning when the string already contains `%`s. The rationale for this is that if the command currently contains format specifiers, then the user is aware that they exist and therefore presumably chose to expand some of the values directly.
